_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+We often want to run Grove as a UI working alongside a [MarkLogic Data Hub](https://marklogic.github.io/marklogic-data-hub/). This guide provides the mechanics on how to do this, and provides a choice of method.
 
-In order to run a Grove UI project alongside the MarkLogic Data Hub Framework (DHF) project, we recommend creating a Grove project-specific app-server, which has its own modules database but points at the content database with the data you wish to visualize: most often the DHF project's FINAL database. The DHF project will be responsible for managing the content database (including setting up indexes - needed for facets - and security permissions), as well as the content database's related triggers and schemas databases, while the Grove project will manage the Grove project's specific app-server and modules database.
+This guide has been vetted on both Data Hub 5.0 and the Data Hub Framework 4.3.1. 
+
+## Migrating Grove-related artifacts and code to the Data Hub
+
+In order to run a Grove UI project alongside a MarkLogic Data Hub, we recommend migrating Grove related artifacts and code from its code base into your Data Hub project.  
+
+Note that this implies some coordination between the Data Hub project and Grove even after migration of Grove's core capabilities. If possible, it is easiest to manage both projects within the same code repository. The Grove files should all be within a dedicated parent directory. (For example, `grove/ui` and `grove/middle-tier`. Note that there is nothing magic about the `grove` directory name, so call it whatever you want.)
+
+You can accomplish this recommended approach by moving a subset of files from inside the `marklogic` directory of your Grove project into your Data Hub project. Grove projects generated using the grove-cli's `grove new` command ship with this `marklogic` directory, which contains configuration files for a standalone MarkLogic database, optimized for Grove's sample data set.  Once the relevant files are migrated, the `marklogic` directory can be deleted entirely.  Additionally, an environment variable will need to be updated to reflect the data-hub-FINAL HTTP app server as configured by the Data Hub project.
+
+The advantages of this approach include: 
+- Combining the Grove-related artifacts with the Data Hub simplifies development and deployments.  The Grove code is logically separated inside its own directory under `data-hub/src/main/ui-modules` for clarity and distinction from Data Hub specific modules.
+- If code is maintained together in the same repository, consistent versioning can be ensured.
+- The Grove UI becomes a downstream consumer of the Data Hub.
+
+There are some downsides to this approach:
+- UI specific code will reside in close proximity to backend specific configurations.  This might not be as preferable to some development teams.
+- Any CRUD operations performed by the Grove project will write to the Data Hub's FINAL database.
+
```

Comment:
There is another 'down-side' (arguable). The security setup in Grove is more advanced and strict than you get with data-hub out of the box. The steps below skip over the ml-configs, and security in specific. I think we should at least add a note to encourage users to think carefully about the security needs for data processed in dhf/dhs.

Reply:
'Lack' of security is comes from DHS that has its own approach to security. Not sure how to phrase that in here..

Reply:
Security approach is different, best to drop the approach of grove, and utilize that of DHF/DHS..

Reply:
Added a note per your comment.